server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1197,6 +1202,12 @@ public void run() {
                             return;
                         }
 
+                        if (nodeHealthService.getHealth() == NodeHealthService.Status.UNHEALTHY) {
+                            logger.warn("skip prevoting as local node is not writable: {}",
```

Comment:
A warning here isn't helpful, we should be logging the failure elsewhere so this will simply result in confusion.

```suggestion
                            logger.debug("skip prevoting as local node is not writable: {}",
```

Also, we have this generic `NodeHealthService` but the log message is very specific: `local node is not writeable`. Maybe the `NodeHealthService` should describe the problem rather than returning a simple boolean.